src/Util/ClassExistenceChecker.php

```diff
@@ -0,0 +1,42 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Roave\BetterReflection\Util;
+
+use function class_exists;
+use function interface_exists;
+use function trait_exists;
+
+/**
+ * @internal
+ */
+class ClassExistenceChecker
+{
+    /**
+     * @psalm-assert-if-true class-string $name
+     */
+    public static function exists(string $name): bool
+    {
+        return self::classExists($name) || self::interfaceExists($name) || self::traitExists($name);
+    }
+
+    public static function classExists(string $name): bool
+    {
+        return class_exists($name, false);
+    }
+
+    public static function interfaceExists(string $name): bool
+    {
+        return interface_exists($name, false);
+    }
+
+    /**
+     * @psalm-suppress InvalidNullableReturnType
+     */
+    public static function traitExists(string $name): bool
+    {
+        /** @psalm-suppress NullableReturnStatement */
```

Comment:
Nani?!??

Reply:
```
ERROR: InvalidNullableReturnType - src/Util/ClassExistenceChecker.php:36:55 - The declared return type 'bool' for Roave\BetterReflection\Util\ClassExistenceChecker::traitExists is not nullable, but 'bool|null' contains null (see https://psalm.dev/144)
    public static function traitExists(string $name): bool


ERROR: NullableReturnStatement - src/Util/ClassExistenceChecker.php:38:16 - The declared return type 'bool' for Roave\BetterReflection\Util\ClassExistenceChecker::traitExists is not nullable, but the function returns 'bool|null' (see https://psalm.dev/139)
        return trait_exists($name, false);
```

🤷‍♂️

Reply:
@orklah do you think we should report this in upstream `vimeo/psalm`?

Reply:
:o trait_exists in Psalm is indeed documented as `?bool` since https://github.com/vimeo/psalm/pull/1541

Not sure where this was coming from. @voku do you have an idea?

In any case, yeah, please create an issue, we'll sort out the answer there!